content/docs/examples/advanced-egress/egress-gateway-mtls-origination/index.md

```diff
@@ -0,0 +1,563 @@
+---
+title: Mutual TLS origination by Egress Gateway
+description: Describes how to configure an Egress Gateway to perform mutual TLS origination to external services.
+weight: 45
+keywords: [traffic-management,egress]
+---
+
+The [Configure an Egress Gateway](/docs/examples/advanced-egress/egress-gateway) example describes how to configure
+Istio to direct the egress traffic through a dedicated service called _Egress Gateway_. This examples shows how to
+configure an Egress Gateway to perform mutual TLS origination to external services.
+You will deploy an [NGINX](https://www.nginx.com/) server in your Kubernetes cluster without injecting Istio sidecar
+proxy into it. This server will simulate a host `nginx.example.com` outside the Istio service mesh.
+
+Then you will configure the Egress Gateway to perform mutual TLS with this server.
+Finally, you will direct the traffic from the application pods inside the mesh to the server outside the mesh through
+the Egress Gateway. The Egress Gateway will perform mutual TLS origination with the NGINX server.
+
+## Generate client and server certificates and keys
+
+Generate the certificates and keys in the same way as in the [Securing Gateways with HTTPS](/docs/tasks/traffic-management/secure-ingress/#generate-client-and-server-certificates-and-keys).
+
+1.  Clone the <https://github.com/nicholasjackson/mtls-go-example> repository:
+
+    {{< text bash >}}
+    $ git clone https://github.com/nicholasjackson/mtls-go-example
+    {{< /text >}}
+
+1.  Change directory to the cloned repository:
+
+    {{< text bash >}}
+    $ pushd mtls-go-example
+    {{< /text >}}
+
+1.  Generate the certificates for `nginx.example.com`.
+    Use any password with the following command:
+
+    {{< text bash >}}
+    $ ./generate.sh nginx.example.com <password>
+    {{< /text >}}
+
+    When prompted, select `y` for all the questions.
+
+1.  Move the certificates into `nginx.example.com` directory:
+
+    {{< text bash >}}
+    $ mkdir ~+1/nginx.example.com && mv 1_root 2_intermediate 3_application 4_client ~+1/nginx.example.com
+    {{< /text >}}
+
+1.  Change directory back:
+
+    {{< text bash >}}
+    $ popd
+    {{< /text >}}
+
+## Deploy an NGINX server
+
+1.  Create a namespace `mesh-external` to represent services outside the Istio mesh. Note that the sidecar proxy will
+    not be automatically injected into the pods in this namespace since the automatic sidecar injection was not
+    [enabled](/docs/setup/kubernetes/sidecar-injection/#deploying-an-app) on it.
+
+    {{< text bash >}}
+    $ kubectl create namespace mesh-external
+    {{< /text >}}
+
+1. Create Kubernetes [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to hold the server's and CA
```

Comment:
That's an important point to agree upon. In my opinion, the goal of the examples is to provide users with working scenarios **without going too much into details and explanations**. No need to explain here what is Certificate Authority and how it will be used, or that there are two different secrets. I see it as follows - the user just clicks on the instructions one by one and sees the example working. These examples are like a set of recipes - you just follow them to complete a particular task, you do not have to understand all the details.

Also note that this one is an advanced example, at this point the reader should already be familiar with Istio, no need to go into details regarding each secret/virtual service/destination rule here.

@geeknoid @frankbu could you please provide your opinion here? This one is important, I think.